### PR TITLE
Repurpose DW repository to host ODH DW integration tests

### PR DESCRIPTION
**Pull request by sutaakar**
Repurpose DW repository to host ODH DW integration tests.
Current OpenShift CI PR check doesn't work, need to be adjusted.

## Description

Currently this repository will host Training operator e2e tests. In future I plan to add here additional integration tests.

## How Has This Been Tested?

Tested manually against OpenShift cluster.

## Merge criteria:

- [x] The commits are squashed in a cohesive manner and have meaningful messages.
- [ ] Testing instructions have been added in the PR body (for PRs involving changes that are not immediately obvious).
- [x] The developer has manually tested the changes and verified that the changes work

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -1,156 +1,34 @@
...
+
+.PHONY: install-kind-sh
+install-kind-sh: $(KIND_SH) ## install KinD shell script if not found
+$(KIND_SH): $(LOCALBIN)
+	test -s $(LOCALBIN)/kind.sh || wget --quiet --output-document=$(LOCALBIN)/kind.sh https://raw.githubusercontent.com/project-codeflare/codeflare-operator/main/test/e2e/kind.sh && chmod +x $(LOCALBIN)/kind.sh
+
+.PHONY: setup-kueue
+setup-kueue: ## Set up Kueue for e2e tests.
+	echo "Installing Kueue into the cluster"
+	kubectl apply --server-side -f https://github.com/kubernetes-sigs/kueue/releases/download/${KUEUE_VERSION}/manifests.yaml
```
Comment (astefanutti) [inline Makefile:23]: Should we install from the ODH fork?

File: tests/trainingoperator/support.go [hunk 2]
```diff
@@ -14,16 +14,16 @@ See the License for the specific language governing permissions and
 limitations under the License.
 */
 
-package integration
+package trainingoperator
```
Comment (astefanutti) [inline tests/trainingoperator/support.go:17]: May I ask we rename the directory and package to something like `training` or `kfto`. I just find `trainingoperator` terrible :)

File: tests/trainingoperator/kfto_kueue_sft_test.go [hunk 3]
```diff
@@ -0,0 +1,148 @@
...
+	. "github.com/onsi/gomega"
+	. "github.com/project-codeflare/codeflare-common/support"
+
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	kftov1 "github.com/kubeflow/training-operator/pkg/apis/kubeflow.org/v1"
+)
+
+func PytorchJob(t Test, namespace, name string) func(g gomega.Gomega) *kftov1.PyTorchJob {
```
[the remark this answers is not in the document]
Comment (sutaakar) [reply tests/trainingoperator/kfto_kueue_sft_test.go:32]: Yes, as soon as this PR is merged.

File: tests/trainingoperator/support.go (hunk 2 above)
Comment (sutaakar) [reply tests/trainingoperator/support.go]: Done

File: Makefile (hunk 1 above)
Comment (astefanutti) [reply Makefile:23]: Can't we use something like:
```
$ kubectl apply -k https://github.com/opendatahub-io/kueue/tree/dev/config/rhoai
```

Comment (sutaakar) [reply Makefile:23]: yes, just trying that

Comment (sutaakar) [reply Makefile]: Done.
Removed KinD bits as KinD is currently not usable with Kueue and kfto due to webhooks.

File: Makefile [hunk 4]
```diff
@@ -1,156 +1,34 @@
...
+	echo "Wait for Kueue deployment"
+	kubectl -n kueue-system wait --timeout=300s --for=condition=Available deployments --all
+	echo "Creating Kueue Resources, to be removed once tests creates own Kueue resources"
+	kubectl apply -f kueue-config.yaml
+
+.PHONY: setup-training-operator
+setup-training-operator: ## Set up Training operator for e2e tests.
+	echo "Deploying Training operator"
+	kubectl apply -k "github.com/kubeflow/training-operator/manifests/overlays/standalone"
```
[the remark this answers is not in the document]
Comment (sutaakar) [reply Makefile]: Done.

File: Makefile (hunk 1 above)
Comment (sutaakar) [reply Makefile]: Maybe later it can be added again, once fixed.

File: tests/kfto/kfto_kueue_sft_test.go [hunk 5]
```diff
@@ -0,0 +1,148 @@
...
+
+func PytorchJob(t Test, namespace, name string) func(g gomega.Gomega) *kftov1.PyTorchJob {
+	return func(g gomega.Gomega) *kftov1.PyTorchJob {
+		job, err := t.Client().Kubeflow().KubeflowV1().PyTorchJobs(namespace).Get(t.Ctx(), name, metav1.GetOptions{})
+		g.Expect(err).NotTo(gomega.HaveOccurred())
+		return job
+	}
+}
+
+// s
```
Comment (astefanutti) [inline tests/kfto/kfto_kueue_sft_test.go:40]: I guess this can be removed :)

**Review by astefanutti — APPROVED** [review]

**Comment by astefanutti** [comment]
OK I didn't expect to have this merged directly :D.

**Pull request merged**